### PR TITLE
ex8_ch_3.py

### PR DESCRIPTION
**Pull request by amymhaddad**
Instructions:
Write a program to evenly divide pizzas. Prompt for the number of people, the number of pizzas, and the number of slices per pizza. Ensure that the number of pieces comes out even. Display the number of pieces of pizza each person should get. If there are leftovers, show the number of leftover pieces.

Example Output
       How many people? 8
       How many pizzas do you have? 2
       8 people with 2 pizzas
       Each person gets 2 pieces of pizza.
       There are 0 leftover pieces.

Challenge: Create a variant of the program that prompts for the number of people and the number of pieces each person wants, and calculate how many full pizzas you need to purchase.

### CODE REVIEW

File: ex8_ch_3.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+total_slices = slices_per_pizza * number_of_pizzas
+
+print(f"There are {people} people and {number_of_pizzas} pizzas.")
+
+pizza_slices_per_person = total_slices / people
+print(f"Each person gets {pizza_slices_per_person} pieces of pizza.")
+
+leftover_pieces = total_slices % people
+print(f"There are {int(leftover_pieces)} leftover pieces.")
+
```
Comment (amymhaddad) [inline ex8_ch_3.py]: How do I make the output pluralization correct? Example: Each person gets 2 pieces of pizza. or
Each person gets 1 piece of pizza.

File: ex8_ch_3.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+people = int(input("How many people will eat pizza? "))
```
Comment (paulghaddad) [inline ex8_ch_3.py]: I'd recommend renaming this variable to `number_of_people`.

File: ex8_ch_3.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+number_of_pizzas = int(input("How many pizzas do you have? "))
+slices_per_pizza = int(input("How many slices are in each pizza? "))
+
+total_slices = slices_per_pizza * number_of_pizzas
+
+print(f"There are {people} people and {number_of_pizzas} pizzas.")
+
+pizza_slices_per_person = total_slices / people
```
Comment (paulghaddad) [inline ex8_ch_3.py]: This is an excellent variable name! 💯

File: ex8_ch_3.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+print(f"Each person gets {pizza_slices_per_person} pieces of pizza.")
+
+leftover_pieces = total_slices % people
```
Comment (paulghaddad) [inline ex8_ch_3.py]: Good variable name and good use of the modulo operator to get the remainder!

File: ex8_ch_3.py (hunk 1 above)
Comment (paulghaddad) [reply ex8_ch_3.py]: This is the `if-else` that you're almost at! But here's a sneak peek:

```
if leftover_pieces == 1:
  print(f"Each person gets 1 piece of pizza.")
else:
   print(f"Each person gets {leftover_pieces) pieces of pizza.")
```

File: ex8_ch_3.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+print(f"There are {int(leftover_pieces)} leftover pieces.")
```
Comment (paulghaddad) [inline ex8_ch_3.py]: At this point in the program, `leftover_pieces` is an `int`, not a `string`, so there's no need to use `int()` to convert it to a `int`.

File: ex8_ch_3.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+eaters = int(input("How many people are eating pizza? "))
+number_of_pieces_wanted = int(input("How many pieces of pizza does each person want? "))
+
+total_slices_needed = eaters * number_of_pieces_wanted
+
+slices_in_a_pizza = 12
+
+total_pizzas = total_slices_needed / slices_in_a_pizza
+
+print(f"We'll need {int(total_pizzas)} total pizzas.")
```
Comment (paulghaddad) [inline ex8_ch_3.py]: Same thing here, `total_pieces` is already an `int`, so no need to convert.

**Comment by amymhaddad** [comment]
@paulghaddad This exercise is updated.

**Pull request merged**